provisioner/jwks.go

```diff
@@ -0,0 +1,142 @@
+package provisioner
+
+import (
+	"crypto"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"encoding/pem"
+	"fmt"
+
+	"github.com/pkg/errors"
+	"gopkg.in/square/go-jose.v2"
+)
+
+const (
+	kubernetesProgrammaticAuthorization = "urn:kubernetes:programmatic_authorization"
+	idTokenResponse                     = "id_token"
+	publicSubject                       = "public"
+	wellKnowOidcConfigurationUrl        = "https://%s.%s"
+	jwksUrl                             = "https://%s.%s/openid-configuration/keys.json"
```

Comment:
this URL you should get from `$issuer/.well-known/openid-configuration`

Reply:
No, this is where I set the value for that field in the discovery document.

Reply:
I think Sandors point is the same as mine, why not pass the full URL from APIServerURL here, instead of this injection which assumes a certain structure of the URL.

Similar to what I commented another place, but here you also need the protocol, so `APIServerURL` is truly the right value to use here IMO.

Reply:
fixed